Doctest ALL the things

This is a list of all files that need doctests in the manual. If you have a PR in progress to doctest a file please claim it here. This is in order for people not to do double work.  If there is a file that you know of that is fully doctests, please mention it as well.

- [x] Introduction] -  @fredrikekre #20183 
- [x] Getting Started - @fredrikekre #20183 
- [x] Variables - @fredrikekre #20183 
- [x] Integers and Floating-Point Numbers - @fredrikekre #20183 
- [x] Mathematical Operations and Elementary Functions - @fredrikekre #20183 
- [x] Complex and Rational Numbers - @fredrikekre #20183 
- [x] Strings - @fredrikekre #20183 
- [x] Functions - @fredrikekre #20201
- [x] Control Flow - @fredrikekre #20202 
- [x] Scope of Variables - @fredrikekre #20202 
- [x] Types - @fredrikekre #20224 
- [x] Methods - @fredrikekre #20224 
- [x] Constructors - @kshyatt #20285
- [ ] Conversion and Promotion - @kshyatt #20297
- [x] Interfaces - @fredrikekre #20287
- [x] Modules - Nothing runnable
- [ ] Documentation - no doctests but some updates: @kristofferc #20442
- [x] Metaprogramming - @kristofferc #20302
- [x] Multi-dimensional Arrays - #20125
- [x] Linear algebra - 
- [x] Networking and Streams - @kristofferc #20304
- [ ] Parallel Computing - 
- [x] Date and DateTime - @kristofferc #20275
- [x] Running External Programs - @fredrikekre #20315
- [ ] Calling C and Fortran Code - @kristofferc #20440
- [x] Handling Operating System Variation - No runnable examples
- [x] Interacting With Julia - @fredrikekre 
- [x] Embedding Julia - Not doctestable
- [ ] Packages - Not so many doctests here but examples should be run through and output updated
- [ ] Package Development - 
- [ ] Profiling - 
- [ ] Stack Traces - 
- [x] Performance Tips - @kristofferc #20312
- [x] Workflow Tips - 
- [x] Style Guide - @fredrikekre
- [x] Frequently Asked Questions - @kristofferc #20291
- [x] Noteworthy Differences from other Languages - Doees not contain any examples
- [x] Unicode Input - Does not contain any examples

cc @fredrikekre.

Hm, this is the full error

```julia
Documenter: populating indices.
ERROR: LoadError: `makedocs` encountered an error. Terminating build
Stacktrace:
 [1] runner(::Type{Documenter.Builder.RenderDocument}, ::Documenter.Documents.Document) at /home/pkm/julia/julia/doc/deps/v0.6/Documenter/src/Builder.jl:202
 [2] dispatch(::Type{Documenter.Builder.DocumentPipeline}, ::Documenter.Documents.Document) at /home/pkm/julia/julia/doc/deps/v0.6/Documenter/src/Selectors.jl:164
 [3] cd(::Documenter.##2#3{Documenter.Documents.Document}, ::String) at ./file.jl:69
 [4] #makedocs#1(::Bool, ::Array{Any,1}, ::Function) at /home/pkm/julia/julia/doc/deps/v0.6/Documenter/src/Documenter.jl:159
 [5] (::Documenter.#kw##makedocs)(::Array{Any,1}, ::Documenter.#makedocs) at ./<missing>:0
 [6] include_from_node1(::String) at ./loading.jl:539
 [7] include(::String) at ./sysimg.jl:14
 [8] process_options(::Base.JLOptions) at ./client.jl:304
 [9] _start() at ./client.jl:370
while loading /home/pkm/julia/julia/doc/make.jl, in expression starting on line 115
Makefile:54: recipe for target 'doctest' failed
make: *** [doctest] Error 1
make: Leaving directory '/home/pkm/julia/julia/doc'
```
is this simply because there were doctest errors?

--- a/doc/src/manual/packages.md
+++ b/doc/src/manual/packages.md
@@ -24,7 +24,7 @@ No packages installed.
 ```
 
 Your package directory is automatically initialized the first time you run a `Pkg` command
-that expects it to exist – which includes  [`Pkg.status()`](@ref). Here's an example non-trivial
+that expects it to exist – which includes [`Pkg.status()`](@ref). Here's an example non-trivial
 set of required and additional packages:
 
 ```julia
@@ -161,7 +161,7 @@ have to run [`Pkg.edit()`](@ref) again to fix the files contents yourself.
 Because the package manager uses libgit2 internally to manage the package git repositories, users
 may run into protocol issues (if behind a firewall, for example), when running [`Pkg.add()`](@ref).
 By default, all GitHub-hosted packages wil be accessed via 'https'; this default can be modified
-by calling [`Pkg.setprotocol!()`](@ref).  The following command can be run from the command line
+by calling [`Pkg.setprotocol!()`](@ref). The following command can be run from the command line
 in order to tell git to use 'https' instead of the 'git' protocol when cloning all repositories,
 wherever they are hosted:
 
@@ -473,7 +473,7 @@ GitHub "home page," find the file (e.g., `README.md`) within the repository's fo
 and click on it. You'll see the contents displayed, along with a small "pencil" icon in the upper
 right hand corner. Clicking that icon opens the file in edit mode. Make your changes, write a
 brief summary describing the changes you want to make (this is your *commit message*), and then
-hit "Propose file change."  Your changes will be submitted for consideration by the package owner(s)
+hit "Propose file change." Your changes will be submitted for consideration by the package owner(s)
 and collaborators.
 
 For larger documentation changes--and especially ones that you expect to have to update in response
@@ -484,7 +484,7 @@ to feedback--you might find it easier to use the procedure for code changes desc
 #### Executive summary
 
 Here we assume you've already set up git on your local machine and have a GitHub account (see
-above).  Let's imagine you're fixing a bug in the Images package:
+above). Let's imagine you're fixing a bug in the Images package:
 
 ```
 Pkg.checkout("Images")           # check out the master branch
@@ -511,10 +511,10 @@ project's main repository. Because the online repository can't see the code on y
 you first *push* your changes to a publicly-visible location, your own online *fork* of the package
 (hosted on your own personal GitHub account).
 
-Let's assume you already have the `Foo` package installed.  In the description below, anything
+Let's assume you already have the `Foo` package installed. In the description below, anything
 starting with `Pkg.` or `PkgDev.` is meant to be typed at the Julia prompt; anything starting
 with `git` is meant to be typed in [julia's shell mode](@ref man-shell-mode) (or using the shell that comes with
-your operating system).  Within Julia, you can combine these two modes:
+your operating system). Within Julia, you can combine these two modes:
 
 ```julia
 julia> cd(Pkg.dir("Foo"))          # go to Foo's folder
@@ -522,7 +522,7 @@ julia> cd(Pkg.dir("Foo"))          # go to Foo's folder
 shell> git command arguments...    # command will apply to Foo
 ```
 
-Now suppose you're ready to make some changes to `Foo`.  While there are several possible approaches,
+Now suppose you're ready to make some changes to `Foo`. While there are several possible approaches,
 here is one that is widely used:
 
   * From the Julia prompt, type [`Pkg.checkout("Foo")`](@ref). This ensures you're running the latest
@@ -540,10 +540,10 @@ here is one that is widely used:
     If you forget to do this step until after you've already made some changes, don't worry: see
     [more detail about branching](@ref man-branch-post-hoc) below.
   * Make your changes. Whether it's fixing a bug or adding new functionality, in most cases your change
-    should include updates to both the `src/` and `test/` folders.  If you're fixing a bug, add your
+    should include updates to both the `src/` and `test/` folders. If you're fixing a bug, add your
     minimal example demonstrating the bug (on the current code) to the test suite; by contributing
     a test for the bug, you ensure that the bug won't accidentally reappear at some later time due
-    to other changes.  If you're adding new functionality, creating tests demonstrates to the package
+    to other changes. If you're adding new functionality, creating tests demonstrates to the package
     owner that you've made sure your code works as intended.
   * Run the package's tests and make sure they pass. There are several ways to run the tests:
 
@@ -571,7 +571,7 @@ here is one that is widely used:
       * From the shell, type `git push`.  This will add your new commit(s) to the same pull request; you
         should see them appear automatically on the page holding the discussion of your pull request.
 
-    One potential type of change the owner may request is that you squash your commits.  See [Squashing](@ref man-squashing-and-rebasing)
+    One potential type of change the owner may request is that you squash your commits. See [Squashing](@ref man-squashing-and-rebasing)
     below.
 
 ### Dirty packages
@@ -586,9 +586,9 @@ Naturally, this deletes any changes you've made.
 ### [Making a branch *post hoc*](@id man-branch-post-hoc)
 
 Especially for newcomers to git, one often forgets to create a new branch until after some changes
-have already been made.  If you haven't yet staged or committed your changes, you can create a
+have already been made. If you haven't yet staged or committed your changes, you can create a
 new branch with `git checkout -b <newbranch>` just as usual--git will kindly show you that some
-files have been modified and create the new branch for you.  *Your changes have not yet been committed to this new branch*,
+files have been modified and create the new branch for you. *Your changes have not yet been committed to this new branch*,
 so the normal work rules still apply.
 
 However, if you've already made a commit to `master` but wish to go back to the official `master`
@@ -671,7 +671,7 @@ of what your package supports.
 
 You might also have no interest in supporting the development version of Julia. Just as you can
 add a floor to the version you expect your users to be on, you can set an upper bound. In this
-case, you would put `julia 0.x 0.y-` in your `REQUIRE` file.  The `-` at the end of the version
+case, you would put `julia 0.x 0.y-` in your `REQUIRE` file. The `-` at the end of the version
 number means pre-release versions of that specific version from the very first commit. By setting
 it as the ceiling, you mean the code supports everything up to but not including the ceiling version.
 
@@ -680,7 +680,7 @@ and do not want to support the current stable version of Julia. If you choose to
 add `julia 0.y-` to your `REQUIRE`. Just remember to change the `julia 0.y-` to `julia 0.y` in
 your `REQUIRE` file once `0.y` is officially released. If you don't edit the dash cruft you are
 suggesting that you support both the development and stable versions of the same version number!
-That would be madness.  See the [Requirements Specification](@ref) for the full format of `REQUIRE`.
+That would be madness. See the [Requirements Specification](@ref) for the full format of `REQUIRE`.
 
 Lastly, in many cases you may need extra packages for testing. Additional packages which
 are only required for tests should be specified in the `test/REQUIRE` file. This `REQUIRE`
@@ -730,7 +730,7 @@ to follow in naming your package:
 
 ### Generating the package
 
-Suppose you want to create a new Julia package called `FooBar`.  To get started, do `PkgDev.generate(pkg,license)`
+Suppose you want to create a new Julia package called `FooBar`. To get started, do `PkgDev.generate(pkg,license)`
 where `pkg` is the new package name and `license` is the name of a license that the package generator
 knows about:
 
@@ -781,14 +781,14 @@ Date:   Wed Oct 16 17:57:58 2013 -0400
 
 At the moment, the package manager knows about the MIT "Expat" License, indicated by `"MIT"`,
 the Simplified BSD License, indicated by `"BSD"`, and version 2.0 of the Apache Software License,
-indicated by `"ASL"`.  If you want to use a different license, you can ask us to add it to the
+indicated by `"ASL"`. If you want to use a different license, you can ask us to add it to the
 package generator, or just pick one of these three and then modify the `~/.julia/v0.6/PACKAGE/LICENSE.md`
 file after it has been generated.
 
 If you created a GitHub account and configured git to know about it, `PkgDev.generate()` will
-set an appropriate origin URL for you.  It will also automatically generate a `.travis.yml` file
+set an appropriate origin URL for you. It will also automatically generate a `.travis.yml` file
 for using the [Travis](https://travis-ci.org) automated testing service, and an `appveyor.yml`
-file for using [AppVeyor](https://www.appveyor.com).  You will have to enable testing on the Travis
+file for using [AppVeyor](https://www.appveyor.com). You will have to enable testing on the Travis
 and AppVeyor websites for your package repository, but once you've done that, it will already
 have working tests. Of course, all the default testing does is verify that `using FooBar` in Julia
 works.
@@ -808,9 +808,9 @@ foo = readcsv(joinpath(datapath, "foo.csv"))
 ### Making Your Package Available
 
 Once you've made some commits and you're happy with how `FooBar` is working, you may want to get
-some other people to try it out.  First you'll need to create the remote repository and push your
+some other people to try it out. First you'll need to create the remote repository and push your
 code to it; we don't yet automatically do this for you, but we will in the future and it's not
-too hard to figure out [^3].  Once you've done this, letting people try out your code is as simple
+too hard to figure out [^3]. Once you've done this, letting people try out your code is as simple
 as sending them the URL of the published repo – in this case:
 
 ```
@@ -866,8 +866,8 @@ index 0000000..30e525e
 +git://github.com/StefanKarpinski/FooBar.jl.git
 ```
 
-This commit is only locally visible, however.  To make it visible to the Julia community, you
-need to merge your local `METADATA` upstream into the official repo.  The `PkgDev.publish()` command
+This commit is only locally visible, however. To make it visible to the Julia community, you
+need to merge your local `METADATA` upstream into the official repo. The `PkgDev.publish()` command
 will fork the `METADATA` repository on GitHub, push your changes to your fork, and open a pull
 request:
 
@@ -934,7 +934,7 @@ index 0000000..c1cb1c1
 ```
 
 The `PkgDev.tag()` command takes an optional second argument that is either an explicit version
-number object like `v"0.0.1"` or one of the symbols `:patch`, `:minor` or `:major`.  These increment
+number object like `v"0.0.1"` or one of the symbols `:patch`, `:minor` or `:major`. These increment
 the patch, minor or major version number of your package intelligently.
 
 Adding a tagged version of your package will expedite the official registration into METADATA.jl
@@ -1010,7 +1010,7 @@ however, you should also fix the `REQUIRE` file in the current version of the pa
 
 The `~/.julia/v0.6/REQUIRE` file, the `REQUIRE` file inside packages, and the `METADATA` package
 `requires` files use a simple line-based format to express the ranges of package versions which
-need to be installed.  Package `REQUIRE` and `METADATA requires` files should also include the
+need to be installed. Package `REQUIRE` and `METADATA requires` files should also include the
 range of versions of `julia` the package is expected to work with. Additionally, packages can
 include a `test/REQUIRE` file to specify additional packages which are only required for testing.
 

--- a/doc/src/manual/parallel-computing.md
+++ b/doc/src/manual/parallel-computing.md
@@ -44,15 +44,15 @@ Generally it makes sense for `n` to equal the number of CPU cores on the machine
 $ ./julia -p 2
 
 julia> r = remotecall(rand, 2, 2, 2)
-Future(2,1,3,Nullable{Any}())
+Future(2, 1, 4, Nullable{Any}())
 
 julia> s = @spawnat 2 1 .+ fetch(r)
-Future(2,1,6,Nullable{Any}())
+Future(2, 1, 5, Nullable{Any}())
 
 julia> fetch(s)
 2×2 Array{Float64,2}:
- 1.60401  1.50111
- 1.17457  1.15741
+ 1.18526  1.50912
+ 1.16296  1.60607
 ```
 
 The first argument to [`remotecall()`](@ref) is the function to call. Most parallel programming
@@ -73,7 +73,7 @@ but is more efficient.
 
 ```julia
 julia> remotecall_fetch(getindex, 2, r, 1, 1)
-0.10824216411304866
+0.18526337335308085
 ```
 
 Remember that [`getindex(r,1,1)`](@ref) is [equivalent](@ref man-array-indexing) to `r[1,1]`, so this call fetches
@@ -85,14 +85,15 @@ the operation for you:
 
 ```julia
 julia> r = @spawn rand(2,2)
-Future(2,1,4,Nullable{Any}())
+Future(2, 1, 4, Nullable{Any}())
 
 julia> s = @spawn 1 .+ fetch(r)
-Future(3,1,5,Nullable{Any}())
+Future(3, 1, 5, Nullable{Any}())
 
 julia> fetch(s)
-1.10824216411304866 1.13798233877923116
-1.12376292706355074 1.18750497916607167
+2×2 Array{Float64,2}:
+ 1.38854  1.9098
+ 1.20939  1.57158
 ```
 
 Note that we used `1 .+ fetch(r)` instead of `1 .+ r`. This is because we do not know where the
@@ -114,7 +115,7 @@ the Julia prompt:
 
 ```julia
 julia> function rand2(dims...)
-         return 2*rand(dims...)
+           return 2*rand(dims...)
        end
 
 julia> rand2(2,2)
@@ -123,14 +124,14 @@ julia> rand2(2,2)
  1.15119   0.918912
 
 julia> fetch(@spawn rand2(2,2))
-ERROR: On worker 2:
-function rand2 not defined on process 2
+ERROR: RemoteException(2, CapturedException(UndefVarError(Symbol("#rand2"))
+[...]
 ```
 
 Process 1 knew about the function `rand2`, but process 2 did not.
 
 Most commonly you'll be loading code from files or packages, and you have a considerable amount
-of flexibility in controlling which processes load code.  Consider a file, `"DummyModule.jl"`,
+of flexibility in controlling which processes load code. Consider a file, `"DummyModule.jl"`,
 containing the following code:
 
 ```julia
@@ -138,7 +139,7 @@ module DummyModule
 
 export MyType, f
 
-type MyType
+mutable struct MyType
     a::Int
 end
 
@@ -213,17 +214,26 @@ that an object be moved to the local machine. [`@spawn`](@ref) (and a few relate
 also moves data, but this is not as obvious, hence it can be called an implicit data movement
 operation. Consider these two approaches to constructing and squaring a random matrix:
 
-```
-# method 1
-A = rand(1000,1000)
-Bref = @spawn A^2
-...
-fetch(Bref)
+Method 1:
 
-# method 2
-Bref = @spawn rand(1000,1000)^2
-...
-fetch(Bref)
+```julia
+julia> A = rand(1000,1000);
+
+julia> Bref = @spawn A^2;
+
+[...]
+
+julia> fetch(Bref);
+```
+
+Method 2:
+
+```julia
+julia> Bref = @spawn rand(1000,1000)^2;
+
+[...]
+
+julia> fetch(Bref);
 ```
 
 The difference seems trivial, but in fact is quite significant due to the behavior of [`@spawn`](@ref).
@@ -264,13 +274,13 @@ with embedded global references (under `Main` module only) manage globals as fol
   if its value has changed. Also, the cluster does not synchronize global bindings across nodes.
   For example:
 
-```julia
-A = rand(10,10)
-remotecall_fetch(()->foo(A), 2)  # worker 2
-A = rand(10,10)
-remotecall_fetch(()->foo(A), 3)  # worker 3
-A = nothing
-```
+  ```julia
+  A = rand(10,10)
+  remotecall_fetch(()->foo(A), 2) # worker 2
+  A = rand(10,10)
+  remotecall_fetch(()->foo(A), 3) # worker 3
+  A = nothing
+  ```
 
   Executing the above snippet results in `Main.A` on worker 2 having a different value from
   `Main.A` on worker 3, while the value of `Main.A` on node 1 is set to `nothing`.
@@ -299,11 +309,10 @@ julia> let B = B
 
 julia> @spawnat 2 whos();
 
-julia> 	From worker 2:	                             A    800 bytes  10×10 Array{Float64,2}
-	From worker 2:	                          Base               Module
-	From worker 2:	                          Core               Module
-	From worker 2:	                          Main               Module
-
+julia>  From worker 2:                               A    800 bytes  10×10 Array{Float64,2}
+        From worker 2:                            Base               Module
+        From worker 2:                            Core               Module
+        From worker 2:                            Main               Module
 ```
 
 As can be seen, global variable `A` is defined on worker 2, but `B` is captured as a local variable
@@ -320,7 +329,7 @@ function in `count_heads.jl`:
 ```julia
 function count_heads(n)
     c::Int = 0
-    for i=1:n
+    for i = 1:n
         c += rand(Bool)
     end
     c
@@ -331,11 +340,16 @@ The function `count_heads` simply adds together `n` random bits. Here is how we 
 trials on two machines, and add together the results:
 
 ```julia
-@everywhere include("count_heads.jl")
+julia> @everywhere include("count_heads.jl")
 
-a = @spawn count_heads(100000000)
-b = @spawn count_heads(100000000)
-fetch(a)+fetch(b)
+julia> a = @spawn count_heads(100000000)
+Future(2, 1, 6, Nullable{Any}())
+
+julia> b = @spawn count_heads(100000000)
+Future(3, 1, 7, Nullable{Any}())
+
+julia> fetch(a)+fetch(b)
+100001564
 ```
 
 This example demonstrates a powerful and often-used parallel programming pattern. Many iterations
@@ -353,8 +367,8 @@ of processes, we can use a *parallel for loop*, which can be written in Julia us
 like this:
 
 ```julia
-nheads = @parallel (+) for i=1:200000000
-  Int(rand(Bool))
+nheads = @parallel (+) for i = 1:200000000
+    Int(rand(Bool))
 end
 ```
 
@@ -372,8 +386,8 @@ For example, the following code will not work as intended:
 
 ```julia
 a = zeros(100000)
-@parallel for i=1:100000
-  a[i] = i
+@parallel for i = 1:100000
+    a[i] = i
 end
 ```
 
@@ -383,8 +397,8 @@ to get around this limitation:
 
 ```julia
 a = SharedArray{Float64}(10)
-@parallel for i=1:10
-  a[i] = i
+@parallel for i = 1:10
+    a[i] = i
 end
 ```
 
@@ -392,8 +406,8 @@ Using "outside" variables in parallel loops is perfectly reasonable if the varia
 
 ```julia
 a = randn(1000)
-@parallel (+) for i=1:100000
-  f(a[rand(1:end)])
+@parallel (+) for i = 1:100000
+    f(a[rand(1:end)])
 end
 ```
 
@@ -411,8 +425,9 @@ operation called *parallel map*, implemented in Julia as the [`pmap()`](@ref) fu
 we could compute the singular values of several large random matrices in parallel as follows:
 
 ```julia
-M = Matrix{Float64}[rand(1000,1000) for i=1:10]
-pmap(svd, M)
+julia> M = Matrix{Float64}[rand(1000,1000) for i = 1:10];
+
+julia> pmap(svd, M);
 ```
 
 Julia's [`pmap()`](@ref) is designed for the case where each function call does a large amount
@@ -439,8 +454,9 @@ work to processes only when they finish their current tasks.
 As an example, consider computing the singular values of matrices of different sizes:
 
 ```julia
-M = Matrix{Float64}[rand(800,800), rand(600,600), rand(800,800), rand(600,600)]
-pmap(svd, M)
+julia> M = Matrix{Float64}[rand(800,800), rand(600,600), rand(800,800), rand(600,600)];
+
+julia> pmap(svd, M);
 ```
 
 If one process handles both 800×800 matrices and another handles both 600×600 matrices, we will
@@ -514,7 +530,7 @@ A channel can be visualized as a pipe, i.e., it has a write end and read end.
     function foo()
         while true
             data = take!(c1)
-            .......             # process data
+            [...]               # process data
             put!(c2, result)    # write out result
         end
     end
@@ -538,34 +554,34 @@ A channel can be visualized as a pipe, i.e., it has a write end and read end.
     On a closed [`Channel`](@ref), [`put!()`](@ref) will fail. For example:
 
 ```julia
-julia> c=Channel(2);
+julia> c = Channel(2);
 
-julia> put!(c,1)   # `put!` on an open channel succeeds
+julia> put!(c, 1) # `put!` on an open channel succeeds
 1
 
 julia> close(c);
 
-julia> put!(c,2)   # `put!` on a closed channel throws an exception.
+julia> put!(c, 2) # `put!` on a closed channel throws an exception.
 ERROR: InvalidStateException("Channel is closed.",:closed)
-...
+[...]
 ```
 
   * [`take!()`](@ref) and [`fetch()`](@ref) (which retrieves but does not remove the value) on a closed
     channel successfully return any existing values until it is emptied. Continuing the above example:
 
 ```julia
-julia> fetch(c)     # Any number of `fetch` calls succeed.
+julia> fetch(c) # Any number of `fetch` calls succeed.
 1
 
 julia> fetch(c)
 1
 
-julia> take!(c)     # The first `take!` removes the value.
+julia> take!(c) # The first `take!` removes the value.
 1
 
-julia> take!(c)     # No more data available on a closed channel.
+julia> take!(c) # No more data available on a closed channel.
 ERROR: InvalidStateException("Channel is closed.",:closed)
-...
+[...]
 ```
 
 A `Channel` can be used as an iterable object in a `for` loop, in which case the loop runs as
@@ -575,25 +591,27 @@ long as the `Channel` has data or is open. The loop variable takes on all values
 For example, the following would cause the `for` loop to wait for more data:
 
 ```julia
-c=Channel{Int}(10)
-foreach(i->put!(c, i), 1:3)    # add a few entries
-data = [i for i in c]
+julia> c = Channel{Int}(10);
+
+julia> foreach(i->put!(c, i), 1:3) # add a few entries
+
+julia> data = [i for i in c]
 ```
 
 while this will return after reading all data:
 
 ```julia
-julia> c=Channel{Int}(10);
+julia> c = Channel{Int}(10);
 
-julia> foreach(i->put!(c, i), 1:3);    # add a few entries
+julia> foreach(i->put!(c, i), 1:3); # add a few entries
 
-julia> close(c);                       # `for` loops can exit
+julia> close(c);                    # `for` loops can exit
 
 julia> data = [i for i in c]
 3-element Array{Int64,1}:
-1
-2
-3
+ 1
+ 2
+ 3
 ```
 
 Consider a simple example using channels for inter-task communication. We start 4 tasks to process
@@ -603,39 +621,51 @@ a tuple of `job_id` and the simulated time to the results channel. Finally all t
 printed out.
 
 ```julia
-const jobs = Channel{Int}(32)
-const results = Channel{Tuple}(32)
+julia> const jobs = Channel{Int}(32);
 
-function do_work()
-    for job_id in jobs
-        exec_time = rand()
-        sleep(exec_time)                # simulates elapsed time doing actual work
-                                        # typically performed externally.
-        put!(results, (job_id, exec_time))
-    end
-end
+julia> const results = Channel{Tuple}(32);
 
-function make_jobs(n)
-    for i in 1:n
-        put!(jobs, i)
-    end
-end
+julia> function do_work()
+           for job_id in jobs
+               exec_time = rand()
+               sleep(exec_time)                # simulates elapsed time doing actual work
+                                               # typically performed externally.
+               put!(results, (job_id, exec_time))
+           end
+       end;
 
-# feed the jobs channel with "n" jobs
-n = 12
-@schedule make_jobs(n)
+julia> function make_jobs(n)
+           for i in 1:n
+               put!(jobs, i)
+           end
+       end;
 
-# start 4 tasks to process requests in parallel
-for i in 1:4
-    @schedule do_work()
-end
+julia> n = 12;
 
-# print out results
-@elapsed while n > 0
-    job_id, exec_time = take!(results)
-    println("$job_id finished in $(round(exec_time,2)) seconds")
-    n = n - 1
-end
+julia> @schedule make_jobs(n); # feed the jobs channel with "n" jobs
+
+julia> for i in 1:4 # start 4 tasks to process requests in parallel
+           @schedule do_work()
+       end
+
+julia> @elapsed while n > 0 # print out results
+           job_id, exec_time = take!(results)
+           println("$job_id finished in $(round(exec_time,2)) seconds")
+           n = n - 1
+       end
+4 finished in 0.22 seconds
+3 finished in 0.45 seconds
+1 finished in 0.5 seconds
+7 finished in 0.14 seconds
+2 finished in 0.78 seconds
+5 finished in 0.9 seconds
+9 finished in 0.36 seconds
+6 finished in 0.87 seconds
+8 finished in 0.79 seconds
+10 finished in 0.64 seconds
+12 finished in 0.5 seconds
+11 finished in 0.97 seconds
+0.029772311
 ```
 
 The current version of Julia multiplexes all tasks onto a single OS thread. Thus, while tasks
@@ -687,51 +717,62 @@ remote store.
     deserialized [`RemoteChannel`](@ref) object (on any worker), therefore also points to the same
     backing store as the original.
 
-The channels example from above channels example can be modified for interprocess communication,
+The channels example from above can be modified for interprocess communication,
 as shown below.
 
 We start 4 workers to process a single `jobs` remote channel. Jobs, identified by an id (`job_id`),
 are written to the channel. Each remotely executing task in this simulation reads a `job_id`,
-waits for a random amout of time and writes back a tuple of `job_id`, time taken and its own
+waits for a random amount of time and writes back a tuple of `job_id`, time taken and its own
 `pid` to the results channel. Finally all the `results` are printed out on the master process.
 
 ```julia
-addprocs(4)         # add worker processes
+julia> addprocs(4); # add worker processes
 
-const jobs = RemoteChannel(()->Channel{Int}(32))
-const results = RemoteChannel(()->Channel{Tuple}(32))
+julia> const jobs = RemoteChannel(()->Channel{Int}(32));
 
-# define work function everywhere
-@everywhere function do_work(jobs, results)
-    while true
-        job_id = take!(jobs)
-        exec_time = rand()
-        sleep(exec_time)                # simulates elapsed time doing actual work
-        put!(results, (job_id, exec_time, myid()))
-    end
-end
+julia> const results = RemoteChannel(()->Channel{Tuple}(32));
 
-function make_jobs(n)
-    for i in 1:n
-        put!(jobs, i)
-    end
-end
+julia> @everywhere function do_work(jobs, results) # define work function everywhere
+           while true
+               job_id = take!(jobs)
+               exec_time = rand()
+               sleep(exec_time) # simulates elapsed time doing actual work
+               put!(results, (job_id, exec_time, myid()))
+           end
+       end
 
-# feed the jobs channel with "n" jobs
-n = 12
-@schedule make_jobs(n)
+julia> function make_jobs(n)
+           for i in 1:n
+               put!(jobs, i)
+           end
+       end;
 
-# start tasks on the workers to process requests in parallel
-for p in workers()
-    @async remote_do(do_work, p, jobs, results)
-end
+julia> n = 12;
 
-# print out results
-@elapsed while n > 0
-    job_id, exec_time, where = take!(results)
-    println("$job_id finished in $(round(exec_time,2)) seconds on worker $where")
-    n = n - 1
-end
+julia> @schedule make_jobs(n); # feed the jobs channel with "n" jobs
+
+julia> for p in workers() # start tasks on the workers to process requests in parallel
+           @async remote_do(do_work, p, jobs, results)
+       end
+
+julia> @elapsed while n > 0 # print out results
+           job_id, exec_time, where = take!(results)
+           println("$job_id finished in $(round(exec_time,2)) seconds on worker $where")
+           n = n - 1
+       end
+1 finished in 0.18 seconds on worker 4
+2 finished in 0.26 seconds on worker 5
+6 finished in 0.12 seconds on worker 4
+7 finished in 0.18 seconds on worker 4
+5 finished in 0.35 seconds on worker 5
+4 finished in 0.68 seconds on worker 2
+3 finished in 0.73 seconds on worker 3
+11 finished in 0.01 seconds on worker 3
+12 finished in 0.02 seconds on worker 3
+9 finished in 0.26 seconds on worker 5
+8 finished in 0.57 seconds on worker 4
+10 finished in 0.58 seconds on worker 2
+0.055971741
 ```
 
 ## Remote References and Distributed Garbage Collection
@@ -772,7 +813,7 @@ Once finalized, a reference becomes invalid and cannot be used in any further ca
 
 ## [Shared Arrays](@id man-shared-arrays)
 
-Shared Arrays use system shared memory to map the same array across many processes.  While there
+Shared Arrays use system shared memory to map the same array across many processes. While there
 are some similarities to a [DArray](https://github.com/JuliaParallel/DistributedArrays.jl), the
 behavior of a [`SharedArray`](@ref) is quite different. In a [DArray](https://github.com/JuliaParallel/DistributedArrays.jl),
 each process has local access to just a chunk of the data, and no two processes share the same
@@ -781,8 +822,8 @@ entire array.  A [`SharedArray`](@ref) is a good choice when you want to have a 
 data jointly accessible to two or more processes on the same machine.
 
 [`SharedArray`](@ref) indexing (assignment and accessing values) works just as with regular arrays,
-and is efficient because the underlying memory is available to the local process.  Therefore,
-most algorithms work naturally on [`SharedArray`](@ref)s, albeit in single-process mode.  In cases
+and is efficient because the underlying memory is available to the local process. Therefore,
+most algorithms work naturally on [`SharedArray`](@ref)s, albeit in single-process mode. In cases
 where an algorithm insists on an [`Array`](@ref) input, the underlying array can be retrieved
 from a [`SharedArray`](@ref) by calling [`sdata()`](@ref). For other `AbstractArray` types, [`sdata()`](@ref)
 just returns the object itself, so it's safe to use [`sdata()`](@ref) on any `Array`-type object.
@@ -794,12 +835,12 @@ SharedArray{T,N}(dims::NTuple; init=false, pids=Int[])
 ```
 
 which creates an `N`-dimensional shared array of a bits type `T` and size `dims` across the processes specified
-by `pids`.  Unlike distributed arrays, a shared array is accessible only from those participating
+by `pids`. Unlike distributed arrays, a shared array is accessible only from those participating
 workers specified by the `pids` named argument (and the creating process too, if it is on the
 same host).
 
 If an `init` function, of signature `initfn(S::SharedArray)`, is specified, it is called on all
-the participating workers.  You can specify that each worker runs the `init` function on a distinct
+the participating workers. You can specify that each worker runs the `init` function on a distinct
 portion of the array, thereby parallelizing initialization.
 
 Here's a brief example:
@@ -840,7 +881,7 @@ julia> S = SharedArray{Int,2}((3,4), init = S -> S[indexpids(S):length(procs(S))
 ```
 
 Since all processes have access to the underlying data, you do have to be careful not to set up
-conflicts.  For example:
+conflicts. For example:
 
 ```julia
 @sync begin
@@ -865,81 +906,86 @@ q[i,j,t+1] = q[i,j,t] + u[i,j,t]
 In this case, if we try to split up the work using a one-dimensional index, we are likely to run
 into trouble: if `q[i,j,t]` is near the end of the block assigned to one worker and `q[i,j,t+1]`
 is near the beginning of the block assigned to another, it's very likely that `q[i,j,t]` will
-not be ready at the time it's needed for computing `q[i,j,t+1]`.  In such cases, one is better
-off chunking the array manually.  Let's split along the second dimension:
+not be ready at the time it's needed for computing `q[i,j,t+1]`. In such cases, one is better
+off chunking the array manually. Let's split along the second dimension.
+Define a function that returns the `(irange, jrange)` indexes assigned to this worker:
 
 ```julia
-# This function retuns the (irange,jrange) indexes assigned to this worker
-@everywhere function myrange(q::SharedArray)
-    idx = indexpids(q)
-    if idx == 0
-        # This worker is not assigned a piece
-        return 1:0, 1:0
-    end
-    nchunks = length(procs(q))
-    splits = [round(Int, s) for s in linspace(0,size(q,2),nchunks+1)]
-    1:size(q,1), splits[idx]+1:splits[idx+1]
-end
+julia> @everywhere function myrange(q::SharedArray)
+           idx = indexpids(q)
+           if idx == 0 # This worker is not assigned a piece
+               return 1:0, 1:0
+           end
+           nchunks = length(procs(q))
+           splits = [round(Int, s) for s in linspace(0,size(q,2),nchunks+1)]
+           1:size(q,1), splits[idx]+1:splits[idx+1]
+       end
+```
 
-# Here's the kernel
-@everywhere function advection_chunk!(q, u, irange, jrange, trange)
-    @show (irange, jrange, trange)  # display so we can see what's happening
-    for t in trange, j in jrange, i in irange
-        q[i,j,t+1] = q[i,j,t] +  u[i,j,t]
-    end
-    q
-end
+Next, define the kernel:
 
-# Here's a convenience wrapper for a SharedArray implementation
-@everywhere advection_shared_chunk!(q, u) = advection_chunk!(q, u, myrange(q)..., 1:size(q,3)-1)
+```julia
+julia> @everywhere function advection_chunk!(q, u, irange, jrange, trange)
+           @show (irange, jrange, trange)  # display so we can see what's happening
+           for t in trange, j in jrange, i in irange
+               q[i,j,t+1] = q[i,j,t] +  u[i,j,t]
+           end
+           q
+       end
+```
+
+We also define a convenience wrapper for a `SharedArray` implementation
+
+```julia
+julia> @everywhere advection_shared_chunk!(q, u) =
+           advection_chunk!(q, u, myrange(q)..., 1:size(q,3)-1)
 ```
 
 Now let's compare three different versions, one that runs in a single process:
 
 ```julia
-advection_serial!(q, u) = advection_chunk!(q, u, 1:size(q,1), 1:size(q,2), 1:size(q,3)-1)
+julia> advection_serial!(q, u) = advection_chunk!(q, u, 1:size(q,1), 1:size(q,2), 1:size(q,3)-1);
 ```
 
 one that uses [`@parallel`](@ref):
 
 ```julia
-function advection_parallel!(q, u)
-    for t = 1:size(q,3)-1
-        @sync @parallel for j = 1:size(q,2)
-            for i = 1:size(q,1)
-                q[i,j,t+1]= q[i,j,t] + u[i,j,t]
-            end
-        end
-    end
-    q
-end
+julia> function advection_parallel!(q, u)
+           for t = 1:size(q,3)-1
+               @sync @parallel for j = 1:size(q,2)
+                   for i = 1:size(q,1)
+                       q[i,j,t+1]= q[i,j,t] + u[i,j,t]
+                   end
+               end
+           end
+           q
+       end;
 ```
 
 and one that delegates in chunks:
 
 ```julia
-function advection_shared!(q, u)
-    @sync begin
-        for p in procs(q)
-            @async remotecall_wait(advection_shared_chunk!, p, q, u)
-        end
-    end
-    q
-end
+julia> function advection_shared!(q, u)
+           @sync begin
+               for p in procs(q)
+                   @async remotecall_wait(advection_shared_chunk!, p, q, u)
+               end
+           end
+           q
+       end;
 ```
 
 If we create `SharedArray`s and time these functions, we get the following results (with `julia -p 4`):
 
 ```julia
-q = SharedArray{Float64,3}((500,500,500))
-u = SharedArray{Float64,3}((500,500,500))
+julia> q = SharedArray{Float64,3}((500,500,500));
 
-# Run once to JIT-compile
-advection_serial!(q, u)
-advection_parallel!(q, u)
-advection_shared!(q,u)
+julia> u = SharedArray{Float64,3}((500,500,500));
+```
 
-# Now the real results:
+Run the functions once to JIT-compile and [`@time`](@ref) them on the second run:
+
+```julia
 julia> @time advection_serial!(q, u);
 (irange,jrange,trange) = (1:500,1:500,1:499)
  830.220 milliseconds (216 allocations: 13820 bytes)
@@ -1015,30 +1061,30 @@ Thus, a minimal cluster manager would need to:
 
 [`addprocs(manager::FooManager)`](@ref addprocs) requires `FooManager` to implement:
 
-```
+```julia
 function launch(manager::FooManager, params::Dict, launched::Array, c::Condition)
-    ...
+    [...]
 end
 
 function manage(manager::FooManager, id::Integer, config::WorkerConfig, op::Symbol)
-    ...
+    [...]
 end
 ```
 
 As an example let us see how the `LocalManager`, the manager responsible for starting workers
 on the same host, is implemented:
 
-```
+```julia
 struct LocalManager <: ClusterManager
     np::Integer
 end
 
 function launch(manager::LocalManager, params::Dict, launched::Array, c::Condition)
-    ...
+    [...]
 end
 
 function manage(manager::LocalManager, id::Integer, config::WorkerConfig, op::Symbol)
-    ...
+    [...]
 end
 ```
 
@@ -1066,8 +1112,8 @@ before using any of the parallel constructs.
 For every worker launched, the [`launch()`](@ref) method must add a `WorkerConfig` object (with
 appropriate fields initialized) to `launched`
 
-```
-type WorkerConfig
+```julia
+mutable struct WorkerConfig
     # Common fields relevant to all cluster managers
     io::Nullable{IO}
     host::Nullable{AbstractString}
@@ -1090,7 +1136,7 @@ type WorkerConfig
 
     connect_at::Nullable{Any}
 
-    .....
+    [...]
 end
 ```
 
@@ -1261,7 +1307,7 @@ julia> Threads.nthreads()
 The number of threads Julia starts up with is controlled by an environment variable called `JULIA_NUM_THREADS`.
 Now, let's start up Julia with 4 threads:
 
-```julia
+```
 export JULIA_NUM_THREADS=4
 ```
 
@@ -1287,7 +1333,7 @@ julia> Threads.threadid()
 
 Let's work a simple example using our native threads. Let us create an array of zeros:
 
-```julia
+```jldoctest
 julia> a = zeros(10)
 10-element Array{Float64,1}:
  0.0
@@ -1309,9 +1355,9 @@ Julia supports parallel loops using the [`Threads.@threads`](@ref) macro. This m
 in front of a `for` loop to indicate to Julia that the loop is a multi-threaded region:
 
 ```julia
-Threads.@threads for i = 1:10
-    a[i] = Threads.threadid()
-end
+julia> Threads.@threads for i = 1:10
+           a[i] = Threads.threadid()
+       end
 ```
 
 The iteration space is split amongst the threads, after which each thread writes its thread ID

--- a/doc/src/manual/profile.md
+++ b/doc/src/manual/profile.md
@@ -2,7 +2,7 @@
 
 The `Profile` module provides tools to help developers improve the performance of their
 code. When used, it takes measurements on running code, and produces output that helps you understand
-how much time is spent on individual line(s).  The most common usage is to identify "bottlenecks"
+how much time is spent on individual line(s). The most common usage is to identify "bottlenecks"
 as targets for optimization.
 
 `Profile` implements what is known as a "sampling" or [statistical profiler](https://en.wikipedia.org/wiki/Profiling_(computer_programming)).
@@ -11,7 +11,7 @@ captures the currently-running function and line number, plus the complete chain
 that led to this line, and hence is a "snapshot" of the current state of execution.
 
 If much of your run time is spent executing a particular line of code, this line will show up
-frequently in the set of all backtraces.  In other words, the "cost" of a given line--or really,
+frequently in the set of all backtraces. In other words, the "cost" of a given line--or really,
 the cost of the sequence of function calls up to and including this line--is proportional to how
 often it appears in the set of all backtraces.
 
@@ -37,17 +37,17 @@ any alternatives.
 Let's work with a simple test case:
 
 ```julia
-function myfunc()
-    A = rand(100, 100, 200)
-    maximum(A)
-end
+julia> function myfunc()
+           A = rand(100, 100, 200)
+           maximum(A)
+       end
 ```
 
 It's a good idea to first run the code you intend to profile at least once (unless you want to
 profile Julia's JIT-compiler):
 
 ```julia
-julia> myfunc()  # run once to force compilation
+julia> myfunc() # run once to force compilation
 ```
 
 Now we're ready to profile this function:
@@ -73,20 +73,20 @@ julia> Profile.print()
                   11 reduce.jl; max; line: 37
 ```
 
-Each line of this display represents a particular spot (line number) in the code.  Indentation
+Each line of this display represents a particular spot (line number) in the code. Indentation
 is used to indicate the nested sequence of function calls, with more-indented lines being deeper
-in the sequence of calls.  In each line, the first "field" indicates the number of backtraces
+in the sequence of calls. In each line, the first "field" indicates the number of backtraces
 (samples) taken *at this line or in any functions executed by this line*. The second field is
 the file name, followed by a semicolon; the third is the function name followed by a semicolon,
-and the fourth is the line number.  Note that the specific line numbers may change as Julia's
+and the fourth is the line number. Note that the specific line numbers may change as Julia's
 code changes; if you want to follow along, it's best to run this example yourself.
 
 In this example, we can see that the top level is `client.jl`'s `_start` function. This is the
-first Julia function that gets called when you launch Julia.  If you examine line 373 of `client.jl`,
+first Julia function that gets called when you launch Julia. If you examine line 373 of `client.jl`,
 you'll see that (at the time of this writing) it calls `run_repl()`, mentioned on the second line.
 This in turn calls `eval_user_input()`. These are the functions in `client.jl` that interpret
 what you type at the REPL, and since we're working interactively these functions were invoked
-when we entered `@profile myfunc()`.  The next line reflects actions taken in the [`@profile`](@ref)
+when we entered `@profile myfunc()`. The next line reflects actions taken in the [`@profile`](@ref)
 macro.
 
 The first line shows that 23 backtraces were taken at line 373 of `client.jl`, but it's not that
@@ -179,7 +179,7 @@ dumbsum3
             dumbsum(1)
 ```
 
-Consequently, this child function gets 3 counts, even though the parent only gets one.  The "tree"
+Consequently, this child function gets 3 counts, even though the parent only gets one. The "tree"
 representation makes this much clearer, and for this reason (among others) is probably the most
 useful way to view the results.
 
@@ -213,16 +213,16 @@ Let's discuss these arguments in order:
     ```
   * The first keyword argument, `format`, was introduced above. The possible choices are `:tree` and
     `:flat`.
-  * `C`, if set to `true`, allows you to see even the calls to C code.  Try running the introductory
+  * `C`, if set to `true`, allows you to see even the calls to C code. Try running the introductory
     example with `Profile.print(C = true)`. This can be extremely helpful in deciding whether it's
     Julia code or C code that is causing a bottleneck; setting `C=true` also improves the interpretability
     of the nesting, at the cost of longer profile dumps.
   * Some lines of code contain multiple operations; for example, `s += A[i]` contains both an array
-    reference (`A[i]`) and a sum operation.  These correspond to different lines in the generated
+    reference (`A[i]`) and a sum operation. These correspond to different lines in the generated
     machine code, and hence there may be two or more different addresses captured during backtraces
-    on this line.  `combine=true` lumps them together, and is probably what you typically want, but
+    on this line. `combine=true` lumps them together, and is probably what you typically want, but
     you can generate an output separately for each unique instruction pointer with `combine=false`.
-  * `cols` allows you to control the number of columns that you are willing to use for display.  When
+  * `cols` allows you to control the number of columns that you are willing to use for display. When
     the text would be wider than the display, you might see output like this:
 
     ```
@@ -279,26 +279,26 @@ on the author's laptop).
 
 # Memory allocation analysis
 
-One of the most common techniques to improve performance is to reduce memory allocation.  The
+One of the most common techniques to improve performance is to reduce memory allocation. The
 total amount of allocation can be measured with [`@time`](@ref) and [`@allocated`](@ref), and
 specific lines triggering allocation can often be inferred from profiling via the cost of garbage
-collection that these lines incur.  However, sometimes it is more efficient to directly measure
+collection that these lines incur. However, sometimes it is more efficient to directly measure
 the amount of memory allocated by each line of code.
 
 To measure allocation line-by-line, start Julia with the `--track-allocation=<setting>` command-line
 option, for which you can choose `none` (the default, do not measure allocation), `user` (measure
 memory allocation everywhere except Julia's core code), or `all` (measure memory allocation at
-each line of Julia code). Allocation gets measured for each line of compiled code.  When you quit
+each line of Julia code). Allocation gets measured for each line of compiled code. When you quit
 Julia, the cumulative results are written to text files with `.mem` appended after the file name,
-residing in the same directory as the source file.  Each line lists the total number of bytes
+residing in the same directory as the source file. Each line lists the total number of bytes
 allocated. The `Coverage` package contains some elementary analysis tools, for example to sort
 the lines in order of number of bytes allocated.
 
-In interpreting the results, there are a few important details.  Under the `user` setting, the
+In interpreting the results, there are a few important details. Under the `user` setting, the
 first line of any function directly called from the REPL will exhibit allocation due to events
-that happen in the REPL code itself.  More significantly, JIT-compilation also adds to allocation
+that happen in the REPL code itself. More significantly, JIT-compilation also adds to allocation
 counts, because much of Julia's compiler is written in Julia (and compilation usually requires
-memory allocation).  The recommended procedure is to force compilation by executing all the commands
+memory allocation). The recommended procedure is to force compilation by executing all the commands
 you want to analyze, then call [`Profile.clear_malloc_data()`](@ref) to reset all allocation counters.
  Finally, execute the desired commands and quit Julia to trigger the generation of the `.mem`
 files.

